<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta content="width=device-width, initial-scale=1.0" name="viewport">

  <title>FastQC</title>
  <meta content="" name="description">
  <meta content="" name="keywords">

  <!-- Favicons -->
  <link href="../assets/img/bioinformatics.png" rel="icon">
  <link href="../assets/img/apple-touch-icon.png" rel="apple-touch-icon">

    <!-- Bootstrap CSS -->
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css">
  <!-- Bootstrap Table of Contents CSS -->
  <link rel="stylesheet" href="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.css"> 

  <!-- Google Fonts -->
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500;1,600;1,700&family=Poppins:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500;1,600;1,700&family=Source+Sans+Pro:ital,wght@0,300;0,400;0,600;0,700;1,300;1,400;1,600;1,700&display=swap" rel="stylesheet">

  <!-- Vendor CSS Files -->
  <link href="../assets/vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">
  <link href="../assets/vendor/bootstrap-icons/bootstrap-icons.css" rel="stylesheet">
  <link href="../assets/vendor/aos/aos.css" rel="stylesheet">
  <link href="../assets/vendor/glightbox/css/glightbox.min.css" rel="stylesheet">
  <link href="../assets/vendor/swiper/swiper-bundle.min.css" rel="stylesheet">

  <!-- Variables CSS Files. Uncomment your preferred color scheme -->
  <link href="../assets/css/variables.css" rel="stylesheet">
  <!-- <link href="assets/css/variables-blue.css" rel="stylesheet"> -->
  <!-- <link href="assets/css/variables-green.css" rel="stylesheet"> -->
  <!-- <link href="assets/css/variables-orange.css" rel="stylesheet"> -->
  <!-- <link href="assets/css/variables-purple.css" rel="stylesheet"> -->
  <!-- <link href="assets/css/variables-red.css" rel="stylesheet"> -->
  <!-- <link href="assets/css/variables-pink.css" rel="stylesheet"> -->

  <!-- Template Main CSS File -->
  <link href="../assets/css/style.css" rel="stylesheet">
  <!-- =======================================================
  * Template Name: HeroBiz
  * Template URL: https://bootstrapmade.com/herobiz-bootstrap-business-template/
  * Updated: Mar 17 2024 with Bootstrap v5.3.3
  * Author: BootstrapMade.com
  * License: https://bootstrapmade.com/license/
  ======================================================== -->
</head>

<body>

  <!-- ======= Header ======= -->
  <header id="header" class="fixed-top">
    <div class="container d-flex align-items-center justify-content-between">

      <!-- <h1 class="logo"><a href="index.html">Bioinformatics Influencer</a></h1> -->
      <!-- Uncomment below if you prefer to use an image logo -->
      <a href="../index.html" class="logo"><img src="../assets/img/logo.png" alt="" class="img-fluid"></a>

      <nav id="navbar" class="navbar">
        <ul>
          <li><a class="nav-link scrollto active" href="../index.html#hero">Home</a></li>
          <li><a class="nav-link scrollto" href="../index.html#services">Services</a></li>
          <li><a class="nav-link scrollto" href="../index.html#about">About</a></li>
          <li><a class="nav-link scrollto o" href="../index.html#portfolio">Portfolio</a></li>
          <li><a class="nav-link scrollto" href="../index.html#contact">Contact</a></li>
          <li><a class="nav-link scrollto" href="../blog/blog.html">Blog</a></li>
          <li><a class="getstarted scrollto" href="../index.html#cta">Get Started</a></li>
        </ul>
        <i class="bi bi-list mobile-nav-toggle"></i>
      </nav><!-- .navbar -->

    </div>
  </header><!-- End Header -->

  <main id="main">

    <!-- ======= Breadcrumbs ======= -->
    <div class="breadcrumbs">
      <div class="container">

        <div class="d-flex justify-content-between align-items-center">

          <ol>
            <li><a href="../index.html">Home</a></li>
            <li><a href="../blog/blog.html">Blog</a></li>
            <li>Quality Control of High-Volume Sequencing Data with FastQC: A Complete Guide</li>
          </ol>
        </div>

      </div>
    </div><!-- End Breadcrumbs -->

    <!-- ======= Blog Details Section ======= -->
    <section id="blog" class="blog" data-spy="scroll" data-target="#toc">
      <div class="container" data-aos="fade-up">

        <div class="row g-5">

          <div class="col-lg-8">

            <article class="blog-details">

              <div class="post-img">
                <img src="../assets/img/blog/blog2/cover.png" alt="" class="img-fluid">
              </div>

              <h2 class="title">Quality Control of High-Volume Sequencing Data with FastQC: A Complete Guide</h2>

              <div class="meta-top">
                <ul>
                  <li class="d-flex align-items-center"><i class="bi bi-person"></i> <a href="../index.html#about">Cyrille Njume</a></li>
                  <li class="d-flex align-items-center"><i class="bi bi-clock"></i> <a href="#"><time datetime="2024-04-23">April 26, 2024</time></a></li>
                </ul>
              </div><!-- End meta top -->

              <div class="content">
                <p>
                  High-throughput sequencing technologies have revolutionized genomics by enabling the generation of massive data volumes in a short time. However, with great power comes great responsibility, and ensuring the quality of the generated data is crucial. This is where FastQC comes into play.
                </p>

                <p>FastQC analyzes a set of sequence files and generates a quality control report for each, comprising various modules to identify potential data issues. If no files are specified, it launches as an interactive graphical application; otherwise, it runs in non-interactive mode, suitable for standardized analysis pipelines.</p>

                <p>FastQC, a Java-based application, offers modular analyses to swiftly highlight potential problems in sequence data. Its key functions include:</p>

                <p>
                  <ol>
                    <li>Importing data from BAM, SAM, or FastQ files (in any variant).</li>
                    <li>Providing an overview to detect potential issues.</li>
                    <li>Generating summary graphs and tables for rapid data assessment.</li>
                    <li>Exporting results to an HTML-based permanent report.</li>
                    <li>Supporting offline operation for automated report generation.</li>
                  </ol>
                </p>

                <p>FastQC is a robust quality control tool tailored for high-throughput sequence data analysis. Written in Java, it requires a compatible Java Runtime Environment (JRE) for operation. The tool utilizes the Picard BAM/SAM Libraries, conveniently included in the download package. FastQC is released under the GPL v3 or later license. Feedback for further improvements is encouraged, with Simon Andrews serving as the initial contact.</p>

                <h3>Installation</h3>
                <p>To begin using FastQC, ensure you have a suitable JRE installed. The tool includes the Picard BAM/SAM Libraries, eliminating the need for separate installations.</p>

                <h4>JRE Installation</h4>
                <p>Download and install a suitable 64-bit JRE, ensuring that the Java application is in your path. On Linux, you can install Java using package managers such as apt or yum.</p>

                <p>Ubuntu / Mint:</p>
                <p><code>sudo apt install default-jre</code></p>

                <p>CentOS / Redhat:</p>
                <p><code>sudo yum install java-1.8.0-openjdk</code></p>

                <p>You can check whether Java is installed by opening the ‘cmd’ program on Windows, or any shell on Linux and typing <code>java -version</code>.</p>

                <p>You should see something like:</p>

                
                  <pre>
                    <code class="python">
                  >java -version
                  openjdk version "11.0.2" 2019–01–15
                  OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+9)
                  OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.2+9, mixed mode)
                    </code>
                  </pre>

                <h4>FastQC Installation</h4>
                
                <p>You have the option to acquire FastQC directly from the Babraham Bioinformatics <a href = "http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">website</a> . The installation steps are simple:</p>

                <p>
                  <ol>
                    <li>Choose the appropriate version according to your operating system.</li>
                    <li>Extract the downloaded file to any location you prefer.</li>
                    <li>If you’re using Linux, you may need to ensure that the fastqc file is executable by executing <code>chmod +x fastqc</code> in your terminal.</li>
                    <li>Finally, launch FastQC by executing the fastqc executable located within the unzipped directory.</li>
                  </ol>
                </p>

                <h3>Quick Start</h3>
                <p>Once FastQC is installed, conducting a basic quality assessment on your sequence data is straightforward. Here’s a step-by-step guide:</p>

                <p>
                  <ol>
                    <li>Open a terminal window (or command prompt for Windows users).</li>
                    <li>Navigate to the directory where your FastQ files are located.</li>
                    <li>Execute the command <code>fastqc data.fastq</code> to initiate the analysis. (Note: SAM/BAM files can also be used).</li>
                  </ol>
                </p>

                <p>FastQC will process the file and produce an HTML report, along with a compressed archive containing the report and related files. You can view the HTML report using any web browser to examine the results.</p>

                <p>Below are examples of commonly used commands in FastQC for customization:</p>

                <p>
                  <ol>
                    <li>Analyzing Multiple Files: If you have multiple FastQ files, you can analyze them simultaneously by listing them after the <code>fastqc</code> command, separated by spaces. For instance: <code>fastqc file1.fastq file2.fastq file3.fastq</code>.</li>
                    <li>Specifying Output Directory: To designate a different directory for output files, utilize the <code>-O</code> option followed by the directory path. Example: <code>fastqc -O /path/to/output/ data.fastq</code>.</li>
                    <li>Skipping ZIP File Creation: By default, FastQC generates a zipped file containing the report and data files. If you only require the HTML report, employ the <code>--noextract</code> option. For instance: <code>fastqc --noextract data.fastq</code>.</li>
                    <li>Adjusting the Number of Threads: FastQC can process multiple files concurrently using multiple threads. Utilize the <code>-t</code> option followed by the desired thread count. Example: <code>fastqc -t 4 file1.fastq file2.fastq</code>.</li>
                    <li>Running in Non-Interactive Mode: If integrating FastQC into a larger automated pipeline, running it in non-interactive mode may be preferable. Utilize the <code>--nogroup</code> option to deactivate the interactive grouping of bases for each sequence. Example: <code>fastqc --nogroup data.fastq</code>.</li>
                  </ol>
                </p>

                <p>For a complete list of commands and options, refer to the official documentation or run <code>fastqc -h</code> in your terminal.</p>

                <p>FastQC is an essential tool for anyone working with high-throughput sequencing data. By quickly assessing data quality, it ensures reliable downstream analyses, making it indispensable for both seasoned bioinformaticians and beginners alike.</p>

                <p>For more details on interpreting FastQC results, visit the <a href = "https://cyrillemesue.medium.com/read-quality-control-and-trimming-in-genomic-analysis-83d8da651f35">Quality Control and Trimming in Genomic Analysis guide by Simon Andrews.</a></p>

                <h4>Reference</h4>

                <p>Andrews, S. (Year). FastQC: A Quality Control Tool for High Throughput Sequence Data [Software]. Available from <a href = "http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
              </div><!-- End post content -->


            </article><!-- End blog post -->

            <div class="post-author d-flex align-items-center">
              <img src="../assets/img/profile.png" class="rounded-circle flex-shrink-0" alt="">
              <div>
                <h4>Cyrille Njume</h4>
                <div class="social-links">
                  <a href="https://twitter.com/CyrilleMesue#"><i class="bi bi-twitter"></i></a>
                  <a href="https://www.facebook.com/profile.php?id=100078245222182"><i class="bi bi-facebook"></i></a>
                  <a href="https://www.instagram.com/cy_njume/"><i class="biu bi-instagram"></i></a>
                  <a href="https://join.skype.com/invite/Io32VYWhujAc" class="google-plus"><i class="bi bi-skype"></i></a>
                  <a href="https://www.linkedin.com/in/cyrillenjume/" class="linkedin"><i class="bi bi-linkedin"></i></a>
                </div>
                <p>
                  Hello! Thanks for taking the time to read my blog post. I'd love to hear your thoughts and insights, so please feel free to share them in the comment section below. Let's connect on social media too!
                </p>
              </div>
            </div><!-- End post author -->

          <giscus-widget
            id="comments"
            repo="CyrilleMesue/Portfolio"
            repoid="R_kgDOH9qPaA"
            category="General"
            categoryid="DIC_kwDOH9qPaM4Ce3Oh"
            mapping="title"
            term="Welcome to giscus!"
            reactionsenabled="1"
            emitmetadata="0"
            inputposition="top"
            theme="light_protanopia"
            lang="en"
            loading="lazy"
          ></giscus-widget>

          </div>

          <div class="col-lg-4">
            
            <div class="sidebar">
              <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div><!-- End Blog Sidebar -->

            <div class="sidebar">
              <div class="sidebar-item recent-posts">
                <h3 class="sidebar-title">Recent Posts</h3>
                <div class="mt-3" id="recentPosts">
                  <!-- Recent posts will be dynamically generated here -->
                </div>
              </div>
            </div><!-- End Blog Sidebar -->


          </div>
        </div>

      </div>
    </section><!-- End Blog Details Section -->

  </main><!-- End #main -->

  <!-- ======= Footer ======= -->
  <footer id="footer" class="footer">
    <div class="footer-legal text-center">
      <div class="container d-flex flex-column flex-lg-row justify-content-center justify-content-lg-between align-items-center">

        <div class="me-md-auto text-center text-md-start">
          <div class="copyright">
            &copy; Copyright <strong><span>Bioinformatics Developer</span></strong>. All Rights Reserved
          </div>
        </div>

        <div class="social-links order-first order-lg-last mb-3 mb-lg-0">
          <a href="https://twitter.com/CyrilleMesue" class="twitter"><i class="bi bi-twitter"></i></a>
          <a href="https://www.facebook.com/profile.php?id=100078245222182" class="facebook"><i class="bi bi-facebook"></i></a>
          <a href="https://www.instagram.com/cy_njume/" class="instagram"><i class="bi bi-instagram"></i></a>
          <a href="https://join.skype.com/invite/Io32VYWhujAc" class="google-plus"><i class="bi bi-skype"></i></a>
          <a href="https://www.linkedin.com/in/cyrillenjume/" class="linkedin"><i class="bi bi-linkedin"></i></a>
        </div>

      </div>
    </div>

  </footer><!-- End Footer -->

  <a href="#" class="scroll-top d-flex align-items-center justify-content-center"><i class="bi bi-arrow-up-short"></i></a>

  <div id="preloader"></div>

  <!-- Vendor JS Files -->
  <script src="../assets/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>
  <script src="../assets/vendor/aos/aos.js"></script>
  <script src="../assets/vendor/glightbox/js/glightbox.min.js"></script>
  <script src="../assets/vendor/isotope-layout/isotope.pkgd.min.js"></script>
  <script src="../assets/vendor/swiper/swiper-bundle.min.js"></script>
  <script src="../assets/vendor/php-email-form/validate.js"></script>
  <!-- jQuery -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
  <!-- Bootstrap JS -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.16.0/umd/popper.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js"></script>
  <!-- Bootstrap Table of Contents JS -->
  <script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script>
  <!-- Template Main JS File -->
  <script src="../assets/js/main.js"></script>


  <script>
    // Load the JSON file
    fetch('https://raw.githubusercontent.com/CyrilleMesue/archives/main/jsons/recent-posts.json')
      .then(response => response.json())
      .then(data => {
        const recentPostsContainer = document.getElementById('recentPosts');

        // Loop through each post and create HTML
        data.forEach(post => {
          const postItem = document.createElement('div');
          postItem.classList.add('post-item', 'mt-3');

          const image = document.createElement('img');
          image.src = post.image;
          image.alt = '';
          image.classList.add('flex-shrink-0');
          postItem.appendChild(image);

          const postDetails = document.createElement('div');
          const title = document.createElement('h4');
          const link = document.createElement('a');
          link.href = post.blogPath;
          link.textContent = post.title;
          title.appendChild(link);
          postDetails.appendChild(title);

          const time = document.createElement('time');
          time.setAttribute('datetime', '2020-01-01'); // Change date if needed
          time.textContent = post.date;
          postDetails.appendChild(time);

          postItem.appendChild(postDetails);

          recentPostsContainer.appendChild(postItem);
        });
      })
      .catch(error => console.error('Error fetching recent posts:', error));
  </script>

<!-- esm.sh -->
<script type="module" src="https://esm.sh/giscus"></script>

</body>

</html>